qcodes/instrument_drivers/agilent/E8527D.py

```diff
@@ -10,26 +10,75 @@
 class Agilent_E8527D(VisaInstrument):
     """
     This is the QCoDeS driver for the Agilent_E8527D signal generator.
-
     This driver will most likely work for multiple Agilent sources.
-
     This driver does not contain all commands available for the E8527D but
     only the ones most commonly used.
     """
     def __init__(self, name: str, address: str,
-                 step_attenuator: bool = False,
                  terminator: str = '\n',
                  **kwargs: Any) -> None:
         super().__init__(name, address, terminator=terminator, **kwargs)
 
+        # Query installed options
+        self._options = self.ask_raw('DIAG:CPU:INFO:OPT:DET?')
+
+        # Determine installed frequency option
+        f_option = None
+        for f_option in ['513', '520', '521', '532', '540', '550', '567']:
+            if f_option in self._options:
+                frequency_option = f_option
+        if f_option == None:
+            raise RuntimeError('Could not determine the frequency option')
```

Comment:
This looks a bit wrong to me. f_option can never be None but frequency_option can be unset.
You could do something like the following:

```python
frequency_option = None
for f_option in ['513', '520', '521', '532', '540', '550', '567']:
    if f_option in self._options:
        frequency_option = f_option
if frequency_option is None:
    raise RuntimeError('Could not determine the frequency option')
```